### DISCUSSION TITLE
Will Linux Mint start using LUKS for home directory encryption in future?

### DISCUSSION THREAD

**Discussion by amirrh6** [linuxmint/discussions#32] [Ideas]
Canonical ('s employee) states eCryptfs is considered [deprecated](https://discourse.ubuntu.com/t/ecryptfs-is-deprecated/11886), and it is recommended to move to LUKS. As far as I know, Linux Mint uses eCryptfs for home directory encryption option inside setup.
Is there any plan on migrating to LUKS?

**Comment by kacperpaczos** [comment]
This is a good idea, I like it. In my opinion, it should also be possible to encrypt a directory when creating a new user on an already installed system, and also to switch from an encrypted directory to an unencrypted one and vice versa. Currently, only the directory of the first user can be encrypted (during installation) and this cannot be undone.

**Comment by billyswong** [comment]
But LUKS requires a separate, fixed-size partition as far as I know? This will not be a real substitute of eCryptfs. With eCryptfs, one don't need to plan ahead how much space for home folder and how much space for system. We ditched swap partition in favour of swap file. I don't see LUKS partition is a good solution to replace eCryptfs.
